hazelcast/src/test/java/com/hazelcast/map/impl/PartitioningStrategyFactoryTest.java

```diff
@@ -98,16 +112,49 @@ public void whenStrategyForMapAlreadyDefined_getPartitioningStrategy_returnsSame
     // see ExceptionUtil.rethrow for all the details).
     @Test
     public void whenStrategyInstantiationThrowsException_getSamePartitioningStrategy_rethrowsException() {
-        NodeEngine nodeEngine = mock(NodeEngine.class);
-        RuntimeException e = new RuntimeException("expected exception");
-        Mockito.when(nodeEngine.getConfigClassLoader()).thenThrow(e);
+        PartitioningStrategyConfig cfg = new PartitioningStrategyConfig();
+        cfg.setPartitioningStrategyClass("NonExistentPartitioningStrategy");
+
+        // while attempting to get partitioning strategy, ClassNotFound exception will be thrown and wrapped in HazelcastException
+        expectedException.expect(new IsInstanceOf(HazelcastException.class));
```

Comment:
I've written a class `RootCauseMatcher` some time ago, which you can use to check for the original exception in cases like this. It will just call `getCause()` down to the bottom, so it doesn't matter in how many exceptions the original one was wrapped. You can see the usage in `ClientDurableExecutorServiceTest`.